### ISSUE TITLE
Hook up endpoints with basic layout

### ISSUE THREAD

**Issue by sanmont3drepo** [3drepo/3drepo.io#2757]
### Description
Add integration for fetching teamspaces and projects with the new endpoints
`/v5/teamspaces` for getting the list of teamspaces that will populate the select box to switch teamspaces 
`/v5/teamspaces/:teamspace/projects/` for getting the projects that will populate the select box to change projects

### Goals
- [ ] Have the selection combo box of teamspaces with data from the endpoint
- [ ] Have the selection combo box of projects  with data from the endpoint, matching the selected teamspace

### Tasks
- [ ] The logged user info should be fetch at the start (login is done in v4) and should be triggered from src\v5\ui\routes\index.tsx for now. When we have a real integration of v4 and v5 it will be have to be in a common root.
- [ ] Add api service for v5
- [ ] Add redux saga/state/selector for fetching teamspaces in v5
- [ ] Add test for saga using mockup data.
- [ ] Use   for 
- [ ] Fetch teamspaces on load in  src\v5\ui\routes\index.tsx  by triggering an action using the 
- [ ] Add redux saga/state/selector for fetching projects in v5
- [ ] Fetch projects based on the url (try to not refetch if the projects for a teamspace has already been fetched)
- [ ] Use createHooksSelectors for creating the hooks for teamspacesSelectors  and projectsSelectos  as is being used in **src\v5\services\selectorsHooks\currentUserSelectors.hooks.ts** 
- [ ] Use those SelectorsHooks in the the components to present the data.
- [ ] use  createActionsDispatchers for the actions as they are used in **src\v5\services\actionsDispatchers\currentUsersActions.dispatchers.ts** and use those to dispatch actions

**Comment by m1chalkubiak** [comment]
@sanmont3drepo I ran into a problem in which I would need your help. It seems that `actions.helper.ts` doesn't seem to work very well in all cases. First of all, I realized that it will now work for the maximum with two-part names, i.e.

`fetchSuccess` will work fine
but `fetchSuccessStatus` will cause the error:
S2551: Property 'FETCH_SUCCESS_STATUS' does not exist on type 'Constants'. Did you mean 'FETCH_SUCCESSSTATUS'?

Additionally, there is an error with the first letter of the second action name part member. There is a hard-coded letter `S`, so when try to set for `setPending` action, will cause the error:
Property 'SET_PENDING' does not exist on type 'Constants'. Did you mean 'SET_SENDING'?
